vendors/espressif/boards/esp32/ports/wifi/iot_wifi.c

```diff
@@ -263,24 +263,48 @@ BaseType_t WIFI_IsConnected( void )
 
 WIFIReturnCode_t WIFI_Off( void )
 {
+    esp_err_t ret;
+    if ((ret = esp_wifi_deinit()) != ESP_OK) {
```

Comment:
The other WIFI functions uses a semaphore for mutual exclusion. Do we need the same here?

Reply:
Yes, we need the same here. Updated accordingly!